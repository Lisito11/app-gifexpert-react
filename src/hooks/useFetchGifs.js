import {useState, useEffect} from 'react';
import getGifs from '../helpers/getGifs'

export const useFetchGifs = (category) =>{

    const [state, setState] = useState({
        data: [],
        loading: true
    });


    //Aqui es donde se hara la peticion para traer las imagenes
    //No pueden ser async porque esperan algo sincrono por eso usaremos promesas
    useEffect(()=> {

        getGifs(category)
        .then( imgs => {
            setState({
                data: imgs,
                loading: false
            })
        })
    // Se evalua unicamente cuando la categoria cambia
    }, [category]);

    return state;

}