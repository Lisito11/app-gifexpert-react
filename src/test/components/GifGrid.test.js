import React from 'react';
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {

    const category = 'something';

    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });


        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes en useFetchGifs', () => {

        const gifs =[{id:'ABC',url:'https://localfd.com', title:'cualquier cosa'}];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>)
        
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    })
    
    
})
