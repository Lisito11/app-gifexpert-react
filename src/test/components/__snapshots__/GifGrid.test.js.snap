// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> debe de mostrar items cuando se cargan imagenes en useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    something
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localfd.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    something
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
